src/arrays/referencing_array_and_its_elements/medium/q11/MainTest.java

```diff
@@ -0,0 +1,80 @@
+package arrays.referencing_array_and_its_elements.medium.q11;
+
+import global.BaseTest;
+import global.tools.CustomAssertions;
+import global.tools.TestOption;
+import global.variables.Clause;
+import global.variables.clauses.*;
+import global.variables.wrappers.Optional;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.Random;
+import java.util.stream.Stream;
+
+import static global.utils.ArrayUtil.*;
+import static org.junit.Assert.assertEquals;
+
+public class MainTest extends BaseTest {
+    // Java
+
+    public static int n = 6;
+    public static char c = 'b';
+
+    public Clause[] testSentence() {
+        TestOption.isInputTest = true;
+        TestOption.defaultInput = arrayToInput(generateRandomWordArray(n));
+        return new Clause[]{
+                new StringLiteral("Enter " + n + " given names:"),
+                new Optional(new StringLiteral(" ")),
+                new NewLine(),
+                new StringLiteral("There are "),
+                new IntegerLiteral("ans"),
+                new StringLiteral(" names that end with the letter " + c),
+                new Optional(new StringLiteral("."))
+        };
+    }
+
+    public void runMain() {
+        VowelNames.main(new String[0]);
+    }
+
+    public static String randStr() {
+        Random r = new Random();
+        String ans = "";
+        for (int i = 0; i < n; i++) {
+            int up = r.nextInt(20) + 1;
+            String st = "" + (char) ('A' + r.nextInt(26));
+            for (int j = 1; j < up; j++)
+                st += "" + (char) ('a' + r.nextInt(26));
+            ans += st + " ";
+        }
+
+        return ans;
+    }
+
+    static Stream<String> inputProvider() {
```

Comment:
Should add a test case where it is guaranteed that the first letter of some names is the asked for character but not the end letter.